### PR TITLE
Support setting a custom deletewritemode.strategy

### PR DESCRIPTION
**Pull request by ArneKlein**
Hi,

we currently have the scenario where we have to create multi version of the same document based on a key in the value and delete all versions based on the key once we receive a tombstone event.

To support this scenario, it would be great to enable setting a custom "deletewritemodel.strategy" just as we currently can for the normal writemodel.strategy

KAFKA-320

### CODE REVIEW

File: src/main/java/com/mongodb/kafka/connect/sink/MongoSinkTopicConfig.java [hunk 1]
```diff
@@ -172,6 +172,13 @@ public String value() {
   static final String WRITEMODEL_STRATEGY_DEFAULT =
       "com.mongodb.kafka.connect.sink.writemodel.strategy.DefaultWriteModelStrategy";
 
+  public static final String DELETEWRITEMODEL_STRATEGY_CONFIG = "deletewritemodel.strategy";
```
Comment (rozza) [inline src/main/java/com/mongodb/kafka/connect/sink/MongoSinkTopicConfig.java:175]: Rename to `DELETE_WRITEMODEL_STRATEGY_CONFIG` and config to: `delete.writemodel.strategy`

File: src/main/java/com/mongodb/kafka/connect/sink/MongoSinkTopicConfig.java [hunk 2]
```diff
@@ -508,12 +515,7 @@ PostProcessors getPostProcessors() {
 
   public WriteModelStrategy getWriteModelStrategy() {
```
Comment (rozza) [inline src/main/java/com/mongodb/kafka/connect/sink/MongoSinkTopicConfig.java:516]: Seems strange to have two methods with the same name. But having different effects (one caching). I'd be tempted to refactor / rename eg: `getWriteModelStrategyFromConfig`

**Review by rozza — CHANGES_REQUESTED** [review]
Apologies for the delay in reviewing. 

I've added a couple of minor nits for being included into master. 
I would like to see some tests added to test the new logic

**Comment by ArneKlein** [comment]
> Apologies for the delay in reviewing.
> 
> I've added a couple of minor nits for being included into master. I would like to see some tests added to test the new logic

Thank you for the review, I added basically the same tests as for getWriteModelStrategy(). Please let me know if this is sufficient

**Review by rozza — APPROVED** [review]

**Comment by rozza** [comment]
LGTM

**Comment by ArneKlein** [comment]
Any chance that this can get merged?

**Comment by jyemin** [comment]
Note: the latest patch doesn't compile:

```
com/mongodb/kafka/connect/sink/MongoSinkConfigTest.java:812: error: incompatible types: Optional<WriteModelStrategy> cannot be converted to WriteModelStrategy
               cfg.getMongoSinkTopicConfig(TEST_TOPIC).getDeleteWriteModelStrategy();
```

**Pull request merged**